client/src/components/ViewModeToggle/ViewModeToggle.js

```diff
@@ -0,0 +1,255 @@
+
+    switch (activeState) {
+      case 'edit':
+        return i18n._t('Admin.EDIT_MODE', 'Edit mode');
+      case 'preview':
+        return i18n._t('Admin.PREVIEW_MODE', 'Preview mode');
+      default:
+        return i18n._t('Admin.SPLIT_MODE', 'Split mode');
+      }
+  }
+
+  toggle() {
+    this.setState(prevState => ({
+      dropdownOpen: !prevState.dropdownOpen
+    }));
+  }
+
+  /**
+   * Event handler for the split mode button.
+   *
+   * @param {Object} event
+   */
+  handleSplitSelect() {
+    // notify and update the store
+    this.props.onSplitSelect();
+  }
+
+  /**
+   * Event handler for the preview mode button.
+   *
+   * @param {Object} event
+   */
+  handlePreviewSelect() {
+    // notify and update the store
+    this.props.onPreviewSelect();
+  }
+
+  /**
+   * Event handler for the edit mode button.
+   *
+   * @param {Object} event
+   */
+  handleEditSelect() {
+    // notify and update the store
+    this.props.onEditSelect();
+  }
+
+  renderSplitDropdownItem() {
+    const { activeState, splitAvailable, splitIconClass } = this.props;
+
+    const itemClass = classNames(
+      'btn', 'btn-secondary', 'icon-view', 'first',
+      splitIconClass,
+      {
+        'viewMode-selected': (activeState === 'split'),
+        disabled: (!splitAvailable)
+      }
+  );
+
+    return (
+      <DropdownItem
+        type="button"
+        className={itemClass}
+        value="split"
+        onClick={this.handleSplitSelect}
+        id="splitModeButton"
+      >
+        {i18n._t('Admin.SPLIT_MODE', 'Split mode')}
+      </DropdownItem>
+    );
+  }
+
+  renderEditDropDownItem() {
+    const { activeState, editIconClass } = this.props;
+
+    // Highlight if chosen view mode
+    const itemClass = classNames(
+      'btn', 'btn-secondary', 'icon-view', 'last',
+      editIconClass,
+      { 'viewMode-selected': (activeState === 'edit') }
+    );
+
+    return (
+      <DropdownItem
+        type="button"
+        className={itemClass}
+        value="content"
+        onClick={this.handleEditSelect}
+      >
+        {i18n._t('Admin.EDIT_MODE', 'Edit mode')}
+      </DropdownItem>
+    );
+  }
+
+  renderPreviewDropDownItem() {
+    const { activeState, previewIconClass } = this.props;
+
+    // Highlight if chosen view mode
+    const itemClass = classNames('btn', 'btn-secondary', 'icon-view',
+      previewIconClass,
+      { 'viewMode-selected': (activeState === 'preview') }
+    );
+
+    return (
+      <DropdownItem
+        type="button"
+        className={itemClass}
+        value="preview"
+        onClick={this.handlePreviewSelect}
+      >
+        {i18n._t('Admin.PREVIEW_MODE', 'Preview mode')}
+      </DropdownItem>
+    );
+  }
+
+    render() {
+      const {
+        activeState,
+        area,
+        splitAvailable,
+        dropdownToggleProps,
+        dropdownMenuProps
+      } = this.props;
+      let { className } = this.props;
+      className = classNames(className);
```

Comment:
I think this is a little confusing - firstly we dereference a prop then we reassign it, unclear as to whether this reassigns the prop which we shouldn't really be doing, I assume it doesn't but only the local variable. Could you instead create a `const dropdownClassName = classNames(className);`  maybe?